drivers/virtualization/virt_ivshmem.h

```diff
@@ -25,13 +46,27 @@ struct ivshmem_param {
 struct ivshmem {
 	DEVICE_MMIO_RAM;
 	struct pcie_dev *pcie;
+	bool ivshmem_v2;
```

Comment:
#ifdef CONFIG_IVSHMEM_V2

Reply:
Thanks for giving this a futher review.

These were my thoughts when implementation this:

* for completeness, this runtime flag would allow using both ivshmem and ivshmem-v2 concurrently 
* the class of processor that use this functionality are probably less concerned with the memory usage of a few unused functions
* keeping the runtime flag and adding the ifdef makes the implementation a little messy

Do you think this should be runtime configurable, ifdef or both?